test/e2e/symlink_check.go

```
@@ -114,292 +105,217 @@ func findNodeHostnameForPV(t *testing.T, pv *v1.PersistentVolume) string {
+	waitForMetric(prometheusClient, fmt.Sprintf("lso_device_link_mismatch{name=\"%s\", policy=\"None\"}", lvdl.Name), 1)
+	waitForMetric(prometheusClient, fmt.Sprintf("ALERTS{alertname=\"LSODeviceLinkMismatch\", alertstate=\"pending\", persistent_volume=\"%s\"}", lvdl.Name), 1)
+
+	lvdl = updateLVDLPolicy(tc.f, lvdl, localv1.DeviceLinkPolicyCurrentLinkTarget)
+	waitForMetric(prometheusClient, fmt.Sprintf("lso_device_link_mismatch{name=\"%s\", policy=\"None\"}", lvdl.Name), 0)
+	waitForMetric(prometheusClient, fmt.Sprintf("lso_device_link_mismatch{name=\"%s\", policy=\"CurrentLinkTarget\"}", lvdl.Name), 1)
+
+	tc.f.Logf("step 1: setting LVDL policy to PreferredLinkTarget")
+	lvdl = updateLVDLPolicy(tc.f, lvdl, localv1.DeviceLinkPolicyPreferredLinkTarget)
+	lvdl = waitForLVDLLinkTargets(tc.f, lvdl, steps[0].target, steps[0].target)
+	verifyNodeSymlinkTarget(tc.namespace, nodeHostName, pv.Spec.Local.Path, steps[0].target)
 	currentSymlink = lvdl.Status.CurrentLinkTarget
```

golangci-lint (2.12.2)

[error] ineffectual assignment to currentSymlink

(ineffassign)